File: StartUpExample/StartUpExampleJava/app/src/main/java/com/tinlone/startupexamplejava/initializers/SomethingInitializer.java
package com.tinlone.startupexamplejava.initializers;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.startup.Initializer;

import java.util.Collections;
import java.util.List;

public class SomethingInitializer implements Initializer<Void> {

    @NonNull
    @Override
    public Void create(@NonNull Context context) {
        Log.i("loglog", "create: SomethingInitializer");
        return null;
    }

    @NonNull
    @Override
    public List<Class<? extends Initializer<?>>> dependencies() {
        return Collections.emptyList();
    }
}

File: StartUpExample/StartUpExampleKotlin/app/src/main/java/com/tinlone/startupexamplekotlin/initializers/SomethingInitializer.kt
package com.tinlone.startupexamplekotlin.initializers

import android.content.Context
import androidx.startup.Initializer
import com.tinlone.startupexamplekotlin.dependencies.Logger

class SomethingInitializer: Initializer<Unit> {
    override fun create(context: Context): Unit {
        Logger.log("loglog","do something init")
        Logger.log("loglog", "create: SomethingInitializer")
    }

    override fun dependencies(): List<Class<out Initializer<*>>> {
        return listOf(LoggerInitializer::class.java)
    }
}